### ISSUE TITLE
Emotes playing really fast

### ISSUE THREAD

**Issue by MagicHack** [SevenTV/chatterino7#46]
**Describe your issue**

Some emotes are playing really fast in chatterino while they play at normal speed on webchat and the 7tv website.
Like https://7tv.app/emotes/60d77d71d7e0d54212a77b60 and https://7tv.app/emotes/60ae717ab351b8d1c055524a

**Screenshots**

On the left chatterino and the right webchat
https://user-images.githubusercontent.com/24930768/136135603-5f5a7e50-5a8c-4afe-a5f3-713b69b9f7a1.mp4

**OS and Chatterino Version**

Windows 10, Chatterino Nightly 7.3.4 (commit b00bee6)

**Comment by goldbattle** [comment]
This is caused by the browser enforcing a minimum frame time. This basically means that even though `hoboPls` specifies that it is 1ms per frame, the actual framerate is 100ms for frame. This can be found here:
https://source.chromium.org/chromium/chromium/src/+/fe1bca993584d616f52d9088d6b1411537441e9d:third_party/blink/renderer/platform/graphics/deferred_image_decoder.cc;l=353-360;bpv=1;bpt=0
https://bugzilla.mozilla.org/show_bug.cgi?id=1511298#c7
```cpp
base::TimeDelta DeferredImageDecoder::FrameDurationAtIndex(size_t index) const {
  base::TimeDelta duration;
  if (metadata_decoder_)
    duration = metadata_decoder_->FrameDurationAtIndex(index);
  if (index < frame_data_.size())
    duration = frame_data_[index].duration_;
  // Many annoying ads specify a 0 duration to make an image flash as quickly as
  // possible. We follow Firefox's behavior and use a duration of 100 ms for any
  // frames that specify a duration of <= 10 ms. See <rdar://problem/7689300>
  // and <http://webkit.org/b/36082> for more information.
  if (duration <= base::TimeDelta::FromMilliseconds(10))
    duration = base::TimeDelta::FromMilliseconds(100);
  return duration;
}
```


This is fixed by setting the minimum to be 100ms here as in the chromium code:
https://github.com/Chatterino/chatterino2/blob/804bc526b52b5c4996d1d449194074934bc8d529/src/messages/Image.cpp#L156-L157
```cpp
QPixmap::fromImage(image);
int duration = reader.nextImageDelay();
if (duration <= 10)
    duration = 100;
duration = std::max(20, duration);
frames.push_back(Frame<QImage>{image, duration});
```
https://user-images.githubusercontent.com/2222562/149057542-dcca9b86-3bd6-4fe3-ae01-127148758fbc.mp4

This code hasn't changed since @fourtf 's [original commit in 2018](https://github.com/Chatterino/chatterino2/commit/35d462d1f1b992c9b852a4bd57f658bd116dbdf2#diff-24f7e36cd830c753f94fd3d004d9e0f441ea4f8728f02dad6c75904deb67daf6R108-R116).

The question is if this is desirable behavior to have this? Some people might want to have a really fast emote, but this would only be visible on chatterino clients (up to its 20ms per frame enforced limit). Personally, I think enforcing the 100ms limit that the browsers have would be fine and make a consistent experience for people that upload / use emotes.

**Comment by jammehcow** [comment]
Was resolved by #3506, can be closed
